plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/rest/TestTrinoRESTCatalogConnectorTest.java

```diff
@@ -0,0 +1,170 @@
+import static io.trino.tpch.TpchTable.LINE_ITEM;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestTrinoRESTCatalogConnectorTest
+        extends TestIcebergParquetConnectorTest
+{
+    @Override
+    protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
+    {
+        switch (connectorBehavior) {
+            case SUPPORTS_RENAME_SCHEMA:
+            case SUPPORTS_CREATE_VIEW:
+            case SUPPORTS_COMMENT_ON_VIEW:
+            case SUPPORTS_CREATE_MATERIALIZED_VIEW:
+            case SUPPORTS_RENAME_MATERIALIZED_VIEW:
+                return false;
+            default:
+                return super.hasBehavior(connectorBehavior);
+        }
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        File warehouseLocation = Files.newTemporaryFolder();
+        warehouseLocation.deleteOnExit();
+
+        Catalog backend = RESTCatalogTestUtils.backendCatalog(warehouseLocation);
+
+        DelegatingRESTSessionCatalog delegatingCatalog = DelegatingRESTSessionCatalog
+                .builder().delegate(backend).build();
+
+        TestingHttpServer testServer = delegatingCatalog.testServer();
+        testServer.start();
+        closeAfterClass(testServer::stop);
+
+        return IcebergQueryRunner.builder()
+                .setBaseDataDir(Optional.of(warehouseLocation.toPath()))
+                .setIcebergProperties(
+                        ImmutableMap.<String, String>builder()
+                                .put("iceberg.file-format", "PARQUET")
+                                .put("iceberg.catalog.type", "rest")
+                                .put("iceberg.rest.uri", testServer.getBaseUrl().toString())
+                                .buildOrThrow())
+                .setInitialTables(ImmutableList.<TpchTable<?>>builder()
+                        .addAll(REQUIRED_TPCH_TABLES)
+                        .add(LINE_ITEM)
+                        .build())
+                .build();
+    }
+
+    @Override
+    @Test
+    public void testCreateSchemaWithLongName()
+    {
+        throw new SkipException("We're hitting max URI length before hitting the limit of the backing catalog");
+    }
+
+    @Override
+    @Test
+    public void testCreateTableWithLongTableName()
+    {
+        throw new SkipException("We're hitting max URI length before hitting the limit of the backing catalog");
+    }
+
+    @Override
+    @Test
+    public void testRenameTableToLongTableName()
+    {
+        throw new SkipException("We're hitting max URI length before hitting the limit of the backing catalog");
+    }
+
+    @Override
+    @Test
+    public void testAddColumnConcurrently()
+    {
+        throw new SkipException("Concurrent commit not supported");
+    }
+
+    @Override
+    @Test
+    public void testCreateTableAsSelectSchemaNotFound()
+    {
+        throw new SkipException("Backing catalog creates schema automatically.");
```

Comment:
This behavior seems a bug. Please file an issue and leave a TODO comment.

Reply:
This doesn't necessarily have to be a bug. Generally it is up to the particular catalog how namespaces & tables are handled and validated. A `namespace` could be something `shallow` (implicit). When a table `a.b.c.d.tbl` is created, the `a.b.c.d` part is then just treated as the `namespace` and listing all namespaces could mean looking at all the prefixes across tables.

We could most likely update the behavior of the underlying JDBC catalog, but it would require an Iceberg release to use this new behavior here.

Reply:
I understand it's not a bug as Iceberg. However, it's a bug as Trino.

Reply:
I don't think we can do anything on the Trino side to fix this and/or change the behavior. This comes essentially down to implementation details in Iceberg's `JdbcCatalog`. See also my comment around the `JdbcCatalog` behavior from [here](https://github.com/trinodb/trino/pull/13294#discussion_r946843363).

Using the REST catalog with a different backend catalog (it needs to be an actual Iceberg Catalog) would result in the right behavior, but setting up a different backing catalog adds more complexity than necessary.

Reply:
We can confirm schema existence beforehand and throws an exception if it doesn't exist.

Reply:
@ebyhr I agree that this is not desirable behavior and filed [Iceberg#5587](https://github.com/apache/iceberg/issues/5587) to track the improvement.  It's definitely non-standard behavior, but also not a specific requirement of the Catalog interface.  I don't think we want to add additional checks in Trino because it will just introduce additional calls.  Other backing catalogs will perform the check, so this is just a temporary artifact of the implementation used for these tests.